src/storage/mod.rs

```diff
@@ -369,16 +378,20 @@ mod test {
     #[test]
     fn test_compression() {
         let orig = "fn main() {}";
-        let (data, alg) = compress(orig.as_bytes()).unwrap();
-        let blob = Blob {
-            mime: "text/rust".into(),
-            content: data.clone(),
-            path: "main.rs".into(),
-            date_updated: Utc::now(),
-            compression: Some(alg),
-        };
-        test_roundtrip(std::slice::from_ref(&blob));
-        assert_eq!(decompress(data.as_slice(), alg).unwrap(), orig.as_bytes());
+        for alg in CompressionAlgorithm::AVAILABLE {
+            println!("testing algorithm {}", alg);
+
+            let data = compress(orig.as_bytes(), *alg).unwrap();
+            let blob = Blob {
+                mime: "text/rust".into(),
+                content: data.clone(),
+                path: "main.rs".into(),
+                date_updated: Utc::now(),
+                compression: Some(*alg),
+            };
+            test_roundtrip(std::slice::from_ref(&blob));
+            assert_eq!(decompress(data.as_slice(), *alg).unwrap(), orig.as_bytes());
+        }
```

Comment:
This could also be a good place to make sure the buffering works (ensuring it works for compressed files as well as db ones)

Reply:
Hmm? This is not testing buffering.

Reply:
Buffering is still involved within the pipeline though, isn't it?

Reply:
Yeah, but the fact buffering works is already tested elsewhere. We don't need to add such tests every time we access the database.